src/fourier.jl

```diff
@@ -0,0 +1,773 @@
+    fullspec::Bool=false,
+    power_type::String="all",
+    return_auxil::Bool=false)
+    
+    local_show_progress = show_progress
+    if silent
+        local_show_progress = (a) -> a
+    end
+    # Initialize stuff
+    cross = unnorm_cross = unnorm_pds1 = unnorm_pds2 = pds1 = pds2 = nothing
+    n_ave = 0
+    fgt0 = n_bin = freq = nothing
+
+    sum_of_photons1 = sum_of_photons2 = 0
+    common_variance1 = common_variance2 = common_variance = nothing
+
+    for (flux1, flux2) in local_show_progress(zip(flux_iterable1,
+                                                flux_iterable2))
+        if isnothing(flux1) || isnothing(flux2) || all(iszero,flux1) || all(iszero,flux2)
+            continue
+        end
+
+        # Does the flux iterable return the uncertainty?
+        # If so, define the variances
+        variance1 = variance2 = nothing
+        if flux1 isa Tuple
+            flux1, err1 = flux1
+            variance1 = Statistics.mean(err1) ^ 2
+        end
+        if flux2 isa Tuple
+            flux2, err2 = flux2
+            variance2 = Statistics.mean(err2) ^ 2
+        end
+
+        # Only use the variance if both flux iterables define it.
+        if isnothing(variance1) || isnothing(variance2) 
+            variance1 = variance2 = nothing
+        else
+            common_variance1 = sum_if_not_none_or_initialize(common_variance1,
+                                                             variance1)
+            common_variance2 = sum_if_not_none_or_initialize(common_variance2,
+                                                             variance2)
+        end
+
+        n_bin = length(flux1)
+
+        # At the first loop, we define the frequency array and the range of
+        # positive frequency bins (after the first loop, cross will not be
+        # None anymore)
+        if isnothing(cross)
+            freq = fftfreq(n_bin, dt)
+            fgt0 = positive_fft_bins(n_bin)
+        end
+
+        # Calculate the FFTs
+        ft1 = fft(flux1)
+        ft2 = fft(flux2)
+
+        # Calculate the sum of each light curve, to calculate the mean
+        n_ph1 = sum(flux1)
+        n_ph2 = sum(flux2)
+        n_ph = sqrt(n_ph1 * n_ph2)
+
+        # Calculate the unnormalized cross spectrum
+        unnorm_power = ft1 .* conj.(ft2)
+        unnorm_pd1 = unnorm_pd2 = 0
+
+        # If requested, calculate the auxiliary PDSs
+        if return_auxil
+            unnorm_pd1 = real.(ft1 .* conj.(ft1))
+            unnorm_pd2 = real.(ft2 .* conj.(ft2))
+        end
+
+        # Accumulate the sum to calculate the total mean of the lc
+        sum_of_photons1 += n_ph1
+        sum_of_photons2 += n_ph2
+
+        # Take only positive frequencies unless the user wants the full
+        # spectrum
+        if !(fullspec)
+            unnorm_power = keepat!(unnorm_power,fgt0)
+            if return_auxil
+                unnorm_pd1 = keepat!(unnorm_pd1,fgt0)
+                unnorm_pd2 = keepat!(unnorm_pd2,fgt0)
```

Comment:
`keepat!` modifies the vector in-place, no need to reassign the output to itself.
```suggestion
            keepat!(unnorm_power,fgt0)
            if return_auxil
                keepat!(unnorm_pd1,fgt0)
                keepat!(unnorm_pd2,fgt0)
```